lib/imperative-test.js

```diff
@@ -489,11 +491,11 @@ class ImperativeTest extends Test {
     });
   }
 
-  resolves(input, expected) {
+  resolves(input, expected = nothingSymbol) {
     if (typeof input === 'function') input = input();
     return input.then(
       res => {
-        this.strictSame(res, expected);
+        if (expected !== nothingSymbol) this.strictSame(res, expected);
```

Comment:
I think you can avoid passing the function as a first argument at all if this condition isn't true:
```js
return input.then(
  expected !== nothingSymbol
    ? res => {
        this.strictSame(res, expected);
        return res;
      }
    : null,
  e => {
    /*...*/
  }
);
```

Reply:
Yeah, but it makes the code harder to read IMO, I'd prefer to leave it as is.

Reply:
You can extract the `onResolve` argument into a variable, that will simplify it.